neuralop/layers/fno_block.py

```diff
@@ -173,6 +170,9 @@ def __init__(
                 "[instance_norm, group_norm, layer_norm]"
             )
 
+        if self.preactivation:
+            self.forward = self.forward_with_preactivation
```

Comment:
Doesn't overriding the other function make it tricky if the user wanted to switch back? 
I guess they can just re-instantiate

Reply:
Is it likely that users will want to alternate between pre-activation and "post-" activation?

Regardless, I suppose a better control structure would be to let `forward()` dispatch to separate implementations. I'll change that.

Reply:
I'm not sure on this one either : I'm torn between the readability of that VS the extra cost of that check every single call. What do you think?

Reply:
Given forward goes on to call a lot of linear algebra, I think the cost of one conditional at the start of each `FNOBlock` is negligible (unless GPU acceleration makes that cost significantly cheaper somehow).

Reply:
Agreed